zebra-state/src/service/read/block.rs

```diff
@@ -88,3 +92,48 @@ where
         })
         .or_else(|| db.transaction(hash))
 }
+
+/// Returns the [`Utxo`] for [`transparent::OutPoint`], if it exists in the
+/// non-finalized `chain` or finalized `db`.
+///
+/// UTXOs may be returned regardless of whether they have been spent.
+pub fn utxo<C>(chain: Option<C>, db: &ZebraDb, outpoint: transparent::OutPoint) -> Option<Utxo>
+where
+    C: AsRef<Chain>,
+{
+    // # Correctness
+    //
+    // The StateService commits blocks to the finalized state before updating
+    // the latest chain, and it can commit additional blocks after we've cloned
+    // this `chain` variable.
+    //
+    // Since UTXOs are the same in the finalized and non-finalized state,
+    // we check the most efficient alternative first. (`chain` is always in
+    // memory, but `db` stores transactions on disk, with a memory cache.)
+    chain
+        .and_then(|chain| chain.as_ref().created_utxo(&outpoint))
+        .or_else(|| db.utxo(&outpoint).map(|utxo| utxo.utxo))
+}
+
+/// Returns the [`Utxo`] for [`transparent::OutPoint`], if it exists in the
+/// `non_finalized_state` or finalized `db`.
+///
+/// UTXOs may be returned regardless of whether they have been spent.
```

Comment:
Consistency note: other comments say "are".
```suggestion
/// UTXOs are returned regardless of whether they have been spent.
```

Reply:
This was actually a deliberate difference, but you're right - it's not clear what it means.

I explained what actually happens in commit 1d4668432a.